x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/prebuilt_rule_upgrade_notifications.md

```diff
@@ -0,0 +1,331 @@
+
+- [Common technical requirements](./prebuilt_rules_common_info.md#common-technical-requirements).
+
+### Product requirements
+
+Functional requirements for the functionality outlined in this test plan.
+
+- [Common product requirements](./prebuilt_rules_common_info.md#common-product-requirements).
+
+User stories for the Rule Management page:
+
+- User can see a notification callout on the Rule Management page when some of the currently installed prebuilt rules can be upgraded to new versions. The callout encourages the user to keep prebuilt rules up-to-date and regularly upgrade them to their latest versions.
+  - User can dismiss the callout.
+- User can see a Rule Updates tab on the Rule Management page when some of the currently installed prebuilt rules can be upgraded to new versions.
+  - User can see the total number of rules available for upgrade on this tab.
+  - User can click on the tab which opens the Rule Upgrade table.
+
+User stories for the Rule Details page:
+
+- User can see a notification callout on the Rule Details page when the rule shown on this page can be upgraded to a new version. The callout encourages the user to upgrade the rule.
+  - User can open a Rule Upgrade flyout to review updates in the latest rule version and perform the upgrade.
+  - User can't dismiss the callout.
+
+User stories for the Rule Editing page:
+
+- User can see a notification callout on the Rule Editing page when the rule shown on this page can be upgraded to a new version. The callout encourages the user to upgrade the rule before editing it.
+  - User can navigate back to the Rule Details page where they can review updates in the latest rule version and perform the upgrade.
+  - User can't dismiss the callout.
+
+## Scenarios
+
+### Rule upgrade notifications on the Rule Management page
+
+#### **Scenario: User is NOT notified on the Rule Management page when no prebuilt rules are installed**
+
+**Automation**: 1 e2e test with mock rules + 1 integration test with mock rules for the /status endpoint.
+
+```Gherkin
+Given no prebuilt rules are installed in Kibana
+When user opens the Rule Management page
+Then user should NOT see a callout to upgrade prebuilt rules
+And user should NOT see a Rule Updates tab
+And user should NOT see a number of rules available to upgrade
+```
+
+#### **Scenario: User is NOT notified on the Rule Management page when all installed prebuilt rules are up to date**
+
+**Automation**: 1 e2e test with mock rules + 1 integration test with mock rules for the /status endpoint.
+
+```Gherkin
+Given <X> prebuilt rules are installed in Kibana
+And all of them are up to date (no new versions are available)
+When user opens the Rule Management page
+Then user should NOT see a callout to upgrade prebuilt rules
+And user should NOT see a Rule Updates tab
+And user should NOT see a number of rules available to upgrade
+
+Examples:
+  | X    |
+  | 1    |
+  | 100  |
+  | 1250 |
```

Comment:
What's the justification for having multiple examples here and in other similar cases? Is it just scale testing? As far as I'm aware the logic for showing this is basically just if X > 0

Reply:
@dplumlee These numbers are just for example, to better illustrate the scenarios.